ee/tabby-webserver/src/service/cron/mod.rs

```diff
@@ -0,0 +1,97 @@
+mod db;
+
+use std::time::Duration;
+
+use tokio_cron_scheduler::{Job, JobScheduler};
+use tracing::error;
+
+use crate::service::db::{DbConn, JobRun};
+
+async fn new_job_scheduler(jobs: Vec<Job>) -> anyhow::Result<JobScheduler> {
+    let scheduler = JobScheduler::new().await?;
+    for job in jobs {
+        scheduler.add(job).await?;
+    }
+    scheduler.start().await?;
+    Ok(scheduler)
+}
+
+pub fn run_cron(db_conn: DbConn) {
+    tokio::spawn(async move {
+        let Ok(job1) = db::refresh_token_job(db_conn.clone()).await else {
+            error!("failed to create db job");
+            return;
+        };
+        // run every 5 minutes
+        let Ok(job2) = repository_job(db_conn.clone(), "sync".to_owned(), "0 1/5 * * * * *").await
```

Comment:
I want to point out one more time that IMHO the sync job schedules too frequent.